### ISSUE TITLE
paddlehub中 gradual unfreezing问题

### ISSUE THREAD

**Issue by mrhan36** [PaddlePaddle/PaddleFormers#606]
## 版本、环境信息
1）PaddleHub1.6.2，PaddlePaddle1.7.0
2）系统环境：Aistudio
3）使用的迁移训练模型：se_resnext101_32x4d_imagenet

## 问题复现
paddlehub迁移学习时无法 gradual unfreezing，finetune_and_eval过程提示：[ WARNING] - The max op-depth in the network is 335. That results in that can't use the gradual unfreeze finetune strategy.

## 项目在aistudio链接  （注：使用版本2）：
https://aistudio.baidu.com/aistudio/projectdetail/476739 

## 问题
请问 gradual unfreezing只适用于特定预训练模型吗？还是我代码有误，谢谢！

###  谢谢！

**Comment by kinghuin** [comment]
Hi，欢迎反馈问题。请忽略这条warning，我们已在即将发布的新版本中去掉这条不该有的warning

**Comment by mrhan36** [comment]
> Hi，欢迎反馈问题。请忽略这条warning，我们已在即将发布的新版本中去掉这条不该有的warning
## 感谢回答！
## 基于以上运行环境，再请教几个问题：
1. 如果不设置 gradual unfreezing，默认是只训练网络的最后一层（softmax层）吗？还是同时训练靠近输出层的若干层？还是全部层？
2. 我在做实验设计时，想分以下4种情况测试，请问我的这种设计和对应代码是否正确：
（1）不gradual unfreezing；     对应代码："gradual_unfreeze": { "blocks": 0  }
（2）把整个网络分为2块进行gradual unfreezing； 对应代码："gradual_unfreeze": { "blocks": 2  }
（3）整个网络分为3块进行gradual unfreezing； 对应代码："gradual_unfreeze": { "blocks": 3  }
（4）整个网络分为5块进行gradual unfreezing； 对应代码："gradual_unfreeze": { "blocks": 5  }

## 谢谢您！

**Comment by kinghuin** [comment]
1. 您代码中module.context有一个参数叫trainable，如果为True，则所有参数都会参与更新。如果为False，则只有最后一层更新。
2. 都是正确的

**Comment by mrhan36** [comment]
> 1. 您代码中module.context有一个参数叫trainable，如果为True，则所有参数都会参与更新。如果为False，则只有最后一层更新。
> 2. 都是正确的

## 好的，明白了，感谢您耐心解答！！！

**Comment by mrhan36** [comment]
> > 1. 您代码中module.context有一个参数叫trainable，如果为True，则所有参数都会参与更新。如果为False，则只有最后一层更新。
> > 2. 都是正确的
> 
## 谢谢回答！
麻烦您再请教两个个问题：

## 问题1
如果代码如下：
input_dict, output_dict, program = module.context(trainable=False)

那是不是意味着除了softmax层，其他层都是无法finetune的，是不是就意味着以下代码就无效了？
  "discriminative": {
    "blocks": 3,
    "factor": 2.6
  },
  "gradual_unfreeze":
   {
    "blocks": 3   
    }

## 问题2：
如果gradual_unfreeze的blocks=3，那么  "discriminative"的blocks取值是不是必须也是3，discriminitive_blocks的取值必须和gradual_unfreeze的blocks数值一致吗？是否允许以下情况，一个设置为3，一个设置为5？

"discriminative": {
"blocks": 5,
"factor": 2.6
},
"gradual_unfreeze":
{
"blocks": 3
}

## 问题3：
在Aistudio环境中，是否可使用paddlehub进行多卡并行训练？我使用以下代码进行多卡训练，似乎无效。
%set_env CUDA_VISIBLE_DEVICES=4
!export CUDA_VISIBLE_DEVICES=4  
config = hub.RunConfig(
    log_interval=10,
    use_cuda=True,
    num_epoch=20,
    batch_size=32,
    eval_interval=50,
    use_data_parallel=True  
    )

## 问题4：
感觉slanted_triangle和wramup思路一致，在参数设置时，二者可以同时设置吗（如下代码），感觉这样设置冲突了，是否二选一即可？
scheduler = {
  "warmup": 0.1,
  "linear_decay": {
      "start_point": 0.1,
      "end_learning_rate": 0.0
  },
  "slanted_triangle": {
    "cut_fraction": 0.1,
    "ratio": 32
}


## 谢谢！
